main.cpp

```diff
@@ -0,0 +1,90 @@
+#include <stdlib.h>
+#include <string.h>
+#include <chrono>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include "cache.h" //contains all auxillary functions
+#include "plru.h"
+// #include "policy.h"
+
+using namespace std;
+using namespace std::chrono;
+
+#define ll long long
+
+Cache* createCacheInstance(string& policy, ll cs, ll bs, ll sa){
+    
+    // check validity here and exit if invalid
+    if(!strcmp(policy.c_str(), "plru")){
```

Comment:
```suggestion
    if(policy == "plru"){
```
This should be fine. (Btw, checkout : [https://www.rttr.org/](https://www.rttr.org/))